src/Protobuf/Decode.elm

```diff
@@ -726,16 +777,46 @@ tagDecoder =
             )
 
 
-varIntDecoder : Decode.Decoder ( Int, Int )
-varIntDecoder =
+pack : IntOperations int -> Decode.Decoder ( Int, int )
+pack =
+    packWith identity
+
+
+packWith : (int -> a) -> IntOperations int -> Decode.Decoder ( Int, a )
+packWith transform config =
+    varIntDecoder config
+        |> Decode.map (Tuple.mapSecond transform)
+
+
+intDecoder : IntOperations int -> Decoder int
+intDecoder config =
+    pack config |> packedDecoder VarInt
+
+
+sintDecoder : IntOperations int -> Decoder int
+sintDecoder config =
+    packWith config.zagZig config |> packedDecoder VarInt
+
+
+uintDecoder : IntOperations int -> Decoder int
+uintDecoder config =
+    packWith config.toUnsigned config |> packedDecoder VarInt
+
+
+varIntDecoder : IntOperations int -> Decode.Decoder ( Int, int )
+varIntDecoder config =
     Decode.unsignedInt8
         |> Decode.andThen
             (\octet ->
                 if Bitwise.and 0x80 octet == 0x80 then
-                    Decode.map (\( usedBytes, value ) -> ( usedBytes + 1, Bitwise.and 0x7F octet + Bitwise.shiftLeftBy 7 value )) varIntDecoder
+                    Decode.map
+                        (\( usedBytes, value ) ->
+                            ( usedBytes + 1, config.add7Bit (Bitwise.and 0x7F octet) value )
+                        )
+                        (varIntDecoder config)
 
                 else
-                    Decode.succeed ( 1, octet )
+                    Decode.succeed ( 1, config.add7Bit octet config.zero )
```

Comment:
This is to convert a `Int` to an `int`? Can we not just add a method to `IntOperations`?

Reply:
(see my suggestion on the `IntOperations` alias)